packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -393,4 +432,115 @@ export const Table: FC<Props> = memo((props: Props) => {
   );
 });
 
+function useMainData(
```

Comment:
this is basically to first filter out rows which only have values in subcolumns and create new dataframe which should be shown as the "main" values. Second based on whether any row is expanded, collect those sub rows which should be visible into a separate dataframe.